worker/handler/handler.go

```diff
@@ -106,24 +113,72 @@ func NewHandlerManagerSet(opts *config.Config) (hms *HandlerManagerSet, err erro
 	var ihits int64 = 0
 	var misses int64 = 0
 	hms = &HandlerManagerSet{
-		hmMap:      make(map[string]*HandlerManager),
-		regMgr:     rm,
-		pipMgr:     pm,
-		sbFactory:  sf,
-		cacheMgr:   cm,
-		config:     opts,
-		workerDir:  opts.Worker_dir,
-		maxRunners: opts.Max_runners,
-		hhits:      &hhits,
-		ihits:      &ihits,
-		misses:     &misses,
+		hmMap:              make(map[string]*HandlerManager),
+		regMgr:             rm,
+		pipMgr:             pm,
+		sbFactory:          sf,
+		cacheMgr:           cm,
+		config:             opts,
+		workerDir:          opts.Worker_dir,
+		maxRunners:         opts.Max_runners,
+		hhits:              &hhits,
+		ihits:              &ihits,
+		misses:             &misses,
+		avgHandlerMemUsage: opts.Average_handler_memory_usage * 1024,
+		sbHardLimit:        opts.Create_sandbox_hard_limit * 1024,
+		numSbCreating:      0,
 	}
 
+	hms.sbCreatingChan = make(chan bool, 1)
+
 	hms.lru = NewHandlerLRU(hms, opts.Handler_cache_size) //kb
 
+	if hms.avgHandlerMemUsage == 0 {
+		go func() {
+			for true {
+				numHandler := 0
+				totalMemUsage := 0
+				hms.mutex.Lock()
+				for _, hm := range hms.hmMap {
+					for e := hm.handlers.Front(); e != nil; e = e.Next() {
+						h := e.Value.(*Handler)
+						if h.sandbox == nil {
+							continue
+						} else if sbState, err := h.sandbox.State(); err != nil {
+							continue
+						} else if sbState == state.Stopped {
+							continue
+						}
+						numHandler++
+						totalMemUsage += handlerUsage(h)
+					}
+				}
+				hms.mutex.Unlock()
+				if numHandler != 0 {
+					hms.avgHandlerMemUsage = totalMemUsage / numHandler
+				}
+				time.Sleep(time.Second)
```

Comment:
One second seems often.  Can you make it longer (say 10 seconds)?  Also, leave a comment that we should eventually make it configurable.

Reply:
Updated in the new commit, made this configurable and default value be 10 sec.